biorxiv/article_distances/04_biorxiv_article_distances_abstract_only.py

```diff
@@ -0,0 +1,239 @@
+    .sort_values("document", ascending=True)
+    .set_index("document")
+)
+PMC_published.head()
+
+article_distances = cdist(
+    biorxiv_published.loc[PMC_published.index.tolist()].drop(
+        ["document", "published_doi", "pmcoa"], axis=1
+    ),
+    PMC_published.drop(["doi", "journal"], axis=1),
+    "euclidean",
+)
+article_distances.shape
+
+articles_distance_df = (
+    biorxiv_published.loc[PMC_published.index.tolist()]
+    .reset_index()[["document", "pmcid"]]
+    .assign(
+        distance=np.diag(article_distances, k=0), journal=PMC_published.journal.tolist()
+    )
+)
+articles_distance_df.head()
+
+# ## biorxiv -> random paper same journal
+
+PMC_off_published = (
+    pmc_journal_mapped_df.drop("doi", axis=1)
+    .query(f"document not in {biorxiv_published.index.tolist()}")
+    .query(f"journal in {articles_distance_df.journal.unique().tolist()}")
+    .groupby("journal", group_keys=False)
+    .apply(lambda x: x.sample(1, random_state=100))
+)
+PMC_off_published.head()
+
+journal_mapper = {
+    journal: col for col, journal in enumerate(PMC_off_published.journal.tolist())
+}
+list(journal_mapper.items())[0:10]
+
+off_article_dist = cdist(
+    biorxiv_published.loc[PMC_published.index.tolist()]
+    .drop(["document", "published_doi", "pmcoa"], axis=1)
+    .values,
+    PMC_off_published.drop(["document", "journal"], axis=1).values,
+    "euclidean",
+)
+off_article_dist.shape
+
+data = []
+for idx, row in tqdm.tqdm(articles_distance_df.iterrows()):
+    if row["journal"] in journal_mapper:
+        data.append(
+            {
+                "document": row["document"],
+                "pmcid": (
+                    PMC_off_published.query(f"journal=='{row['journal']}'")
+                    .reset_index()
+                    .document.values[0]
+                ),
+                "journal": row["journal"],
+                "distance": off_article_dist[idx, journal_mapper[row["journal"]]],
+            }
+        )
+
+final_df = articles_distance_df.assign(label="pre_vs_published").append(
+    pd.DataFrame.from_records(data).assign(label="pre_vs_random")
+)
+final_df.head()
+
+final_df = biorxiv_journal_df[["document", "preprint_doi"]].merge(final_df)
+final_df.to_csv(
+    "output/annotated_links/article_distances_abstract_only.tsv", sep="\t", index=False
+)
+final_df.head()
+
+# # Distribution plot
+
+g = (
+    p9.ggplot(
+        final_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=1.35)
+)
+g.save("output/figures/biorxiv_article_distance_abstract_only.svg", dpi=250)
+g.save("output/figures/biorxiv_article_distance_abstract_only.png", dpi=250)
```

Comment:
The distances for the abstract_only plots seem to be larger than the ones when looking at the entire article. Is that an artifact of normalization in different text sizes? Or do abstracts change more than the full text when publishing?

I guess since the distance to random articles is also different, would it be possible to normalize based on the mean random article distance so the two plots are more comparable?

Reply:
>The distances for the abstract_only plots seem to be larger than the ones when looking at the entire article. Is that an artifact of normalization in different text sizes? Or do abstracts change more than the full text when publishing?

Based on [this paper](https://www.biorxiv.org/content/10.1101/2021.02.20.432090v2.full.pdf), I'm inclined to say that this is an artifact from the approach I'm using to generate document vectors. Full text has a lot more words than abstracts, so it would be more agnostic to small changes. Hence lower document distances.

>I guess since the distance to random articles is also different, would it be possible to normalize based on the mean random article distance so the two plots are more comparable?

Hmm.. Can't think of any methods off the top of my head, but will see more into this question

Reply:
> Hmm.. Can't think of any methods off the top of my head, but will see more into this question

To follow up on my own comment. I couldn't think of any normalization tricks to make abstracts directly comparable with full text; however, I think the histogram plot gives the desired message that abstracts alone are a viable way to match documents (if you accept the high sensitivity to small changes as a caveat).